perun-core/src/main/java/cz/metacentrum/perun/core/impl/VosManagerImpl.java

```diff
@@ -528,4 +530,49 @@ public boolean hasEmbeddedGroupsItemInForm(PerunSession sess, int voId) {
 					"(SELECT id FROM application_form WHERE vo_id = ? AND group_id IS NULL) AND type = 'EMBEDDED_GROUP_APPLICATION';", voId);
 		return count > 0;
 	}
+
+	@Override
+	public void removeMemberVo(PerunSession sess, Vo vo, Vo memberVo) throws RelationNotExistsException {
+		try {
+			if (0 == jdbc.update("DELETE FROM vos_vos WHERE vo_id = ? AND member_vo_id = ?",
+				vo.getId(), memberVo.getId())) {
+				throw new RelationNotExistsException("Relation between " + vo + " and member vo " + memberVo + " does not exist.");
+			}
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public void addMemberVo(PerunSession sess, Vo vo, Vo memberVo) throws RelationExistsException {
+		try {
+			jdbc.update("INSERT INTO vos_vos(vo_id, member_vo_id, created_at, created_by, " +
+					"modified_at, modified_by) VALUES(?,?," + Compatibility.getSysdate() + ",?," + Compatibility.getSysdate() + ",?)",
+				vo.getId(), memberVo.getId(), sess.getPerunPrincipal().getActor(), sess.getPerunPrincipal().getActor());
+		} catch (DataIntegrityViolationException ex) {
+			throw new RelationExistsException("Relation between " + vo + " and its member " + memberVo + " already exists." , ex);
+		} catch (RuntimeException e) {
+			throw new InternalErrorException(e);
+		}
+	}
+
+	@Override
+	public List<Vo> getMemberVos(PerunSession sess, int voId) throws VoNotExistsException {
```

Comment:
`VoNotExistsException` can now be removed from the method's signature in Impl and BL layers.